### PR TITLE
[Sync -  20240909] Upstream master to ODH master

### PR DESCRIPTION
**Pull request by Jooho**
Regular Sync with upstream kserve master branch to odh master branch.

### CODE REVIEW

File: config/default/kustomization.yaml [hunk 1]
```diff
@@ -204,16 +204,3 @@ patches:
 #   target:
 #     kind: CustomResourceDefinition
 #     name: inferenceservices.serving.kserve.io
-- patch: |-
```
Comment (spolti) [inline config/default/kustomization.yaml:207]: we can remove the commented lines above as well

File: docs/samples/explanation/alibi/alibiexplainer/poetry.lock [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# This file is automatically @generated by Poetry 1.8.3 and should not be changed by hand.
+# This file is automatically @generated by Poetry 1.7.1 and should not be changed by hand.
```
Comment (spolti) [inline docs/samples/explanation/alibi/alibiexplainer/poetry.lock:1]: you need to update your poetry version to 1.8.3 at least.

Comment (Jooho) [reply docs/samples/explanation/alibi/alibiexplainer/poetry.lock:1]: interesting. I removed the poetry and generated it. Why it used old version ??

File: config/default/kustomization.yaml (hunk 1 above)
Comment (Jooho) [reply config/default/kustomization.yaml:207]: removed.

File: docs/samples/explanation/alibi/alibiexplainer/poetry.lock (hunk 2 above)
Comment (Jooho) [reply docs/samples/explanation/alibi/alibiexplainer/poetry.lock]: it is because of local installed poetry version. I updated it and regenerated it.

**Comment by spolti** [comment]
```
failed after 9m7s (failed containers: test): ContainerFailed one or more containers exited
```

Can it be related to the tests needing more time to complete? I didn't spot tests failing.

The other failure, not related with tests as well
```
failed to get CLI image: unable to tag the override 'cli' image into the amd64-cli:latest: ImageStreamTag.image.openshift.io "amd64-cli:latest" is invalid: metadata: Invalid value: "": may not update fields other than metadata.annotations
```

**Comment by israel-hdez** [comment]
Local run:

image

I'll need to look closer at the failures and start fixing... Most of the time, it is something related to our setup, than true-positives.

**Review by israel-hdez — COMMENTED** [review]
I used this for working on https://issues.redhat.com/browse/RHOAIENG-10277, since this sync contains changes required on the ODH setup: the support for dedicated KServe gateways (related to opendatahub-io/opendatahub-operator#1056). Without those changes, odh/master is broken in the ODH setup. So, it made more sense to use this sync.

The new markers that are relevant for ODH E2Es are:
* predictor
* path_based_routing

The other 7 markers are for cases not well supported in ODH. I was able to figure out the needed configs and changes to the codebase to successfully run this markers with 100% of E2Es passing, excluding GRPC-related tests which are not working in ODH.

The previous means that there is confidence that this code sync is working. However, in code review several poetry lockfiles are too different from upstream. IMHO, such large difference is not OK. I would have rejected this PR because of the large differences in poetry lockfiles.

Side finds:
* A review is needed of ODH overlays, as they look updated. I had to fix the overlays to have a working setup.
* OWNER files need to be updated.
* We need to revert the large resources requests on the `manager` that was applied in #261, since upstream improved memory usage in kserve/kserve#3469
* We need to contribute some fixed to the E2Es code to make them compatible with openshift-ci.

In any case, it was already decided to drop this PR in favor on creating a more up-to-date one. So, closing this PR...

**Pull request closed without merging**